mne/viz/epochs.py

```diff
@@ -65,71 +67,252 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
+        in the format (n_epochs, n_channels, n_times). It must return an
+        array (n_epochs, n_times). For example:
+
+            `combine = lambda data: np.median(data, 1)`
+
+    groupby : None | str | dict
+        If not None, combine must not be None. In this case, combining happens
+        over channel groups defined by this parameter.
+        If str, must be "type", in which case one figure per channel type is
+        returned (combining within channel types).
+        If a dict, the values must be picks and one figure will be returned
+        for each entry, aggregating over the corresponding pick groups. This
+        is useful for e.g. ROIs. Each entry must contain only one channel type.
+        For example:
+
+           `groupby = dict(Left_ROI=[1, 2, 3, 4], Right_ROI=[5, 6, 7, 8])`
 
     Returns
     -------
     figs : lists of matplotlib figures
         One figure per channel displayed.
     """
-    from scipy import ndimage
     units = _handle_default('units', units)
     scalings = _handle_default('scalings', scalings)
 
-    import matplotlib.pyplot as plt
+    if groupby is not None and combine is None:
+        raise ValueError("If groupby is not None, combine must not be None.")
+
     if picks is None:
         picks = pick_types(epochs.info, meg=True, eeg=True, ref_meg=False,
-                           exclude='bads')[:5]
+                           exclude='bads')
+        if groupby is None:
+            picks = picks[:5]  # take 5 picks to prevent spawning many figs
+    else:
+        picks = np.atleast_1d(picks)
+
+    if ((fig is not None or axes is not None) and len(picks) > 1
+        and combine is None):
+        raise ValueError('Only single pick can be drawn to a figure/axis; '
+                         'provide only one pick, or use `combine`.')
 
     if set(units.keys()) != set(scalings.keys()):
         raise ValueError('Scalings and units must have the same keys.')
 
-    picks = np.atleast_1d(picks)
-    if (fig is not None or axes is not None) and len(picks) > 1:
-        raise ValueError('Only single pick can be drawn to a figure.')
+    ch_types = [channel_type(epochs.info, idx) for idx in picks]
+    if len(set(ch_types)) > 1 and groupby is None and combine is not None:
+        warn("Combining over multiple channel types. "
+             "Please use `groupby`.")
+    for ch_type in ch_types:
+        if ch_type not in scalings:
+            # We know it's not in either scalings or units since keys match
+            raise KeyError('%s type not in scalings and units' % ch_type)
+
+    if isinstance(axes, dict):
+        show = False
+
+    # First, we collect groupings of picks and types in two lists
+    # (all_picks, all_ch_types, names) -> groupby.
+    # Then, we construct a list of the corresponding data, names and evokeds
+    # (to_plot_list) -> combine.
+    # Then, we loop over this list and plot using _plot_epochs_image.
```

Comment:
The description above is a bit redundant with the code and comments below. I think it would be clearer if the comments were above the code that does what description says.